cmd/ndm_daemonset/probe/udevprobe.go

```diff
@@ -142,20 +144,27 @@ func (up *udevProbe) scan() error {
 			continue
 		}
 		if newUdevice.IsDisk() || newUdevice.IsParitition() {
-			uuid := newUdevice.GetUid()
-			disksUid = append(disksUid, uuid)
 			deviceDetails := &blockdevice.BlockDevice{}
-			deviceDetails.UUID = uuid
+			if up.controller.FeatureGates.IsEnabled(features.GPTBasedUUID) {
+				// all fields that is used to generate UUID to be filled whenever the event is
+				// generated. This is required so that even if none of the probes work, the
+				// system should be able to generate the UUID of the blockdevice
```

Comment:
Comment is not clear to me sorry.

Reply:
`WWN`, `PartitionTableUUID/GPTLabel`, `PartitionUUID`, `FileSystemUUID` and `DeviceType` are the fields we use to generate the UUID. These fields will be fetched from the udev event itself. This is to guarantee that we donot need to rely on any other probes to fill in those details which are ciritcal for device identification.

Reply:
May I suggest you replace what you just said with the code comment?